src/commands/github/actionCommands.ts

```diff
@@ -34,4 +39,30 @@ export async function cancelAction(context: IActionContext, node?: ActionTreeIte
     const client: Octokit = await createOctokitClient();
     await client.actions.cancelWorkflowRun({ owner: node.data.repository.owner.login, repo: node.data.repository.name, run_id: node.data.id });
     await node.refresh(); // need to refresh to update the data
+    await checkActionStatus(context, node);
+}
+
+async function checkActionStatus(context: IActionContext, node: ActionTreeItem): Promise<void> {
+    const startTime: number = Date.now();
+    const client: Octokit = await createOctokitClient();
+    const pollingOperation: () => Promise<boolean> = async () => {
+        const workflowRun: ActionsGetWorkflowRunResponseData = (await client.actions.getWorkflowRun({ owner: node.data.repository.owner.login, repo: node.data.repository.name, run_id: node.data.id })).data;
+        if (ensureStatus(workflowRun) === 'completed') {
+            const actionCompleted: string = localize('actionCompleted', 'Action "{0}" has completed with the conclusion "{1}".', node.data.id, workflowRun.conclusion);
+            ext.outputChannel.appendLog(actionCompleted);
+            window.showInformationMessage(actionCompleted);
+            await node.refresh();
+            context.telemetry.properties.secToReport = String((Date.now() - startTime) / 1000);
+            context.telemetry.properties.conclusion = workflowRun.conclusion;
+            return true;
+        }
+
+        return false;
+    };
+
+    if (!await pollAsyncOperation(pollingOperation, 15, 20 * 60, String(node.data.id))) {
```

Comment:
Use `nonNullProp` instead of `String()`

Or just use `node.fullId` which can't be undefined

Reply:
The issue is that node.data.id is a number, not a string.

But since it's just for keeping track of cancellation tokens, I think node.fullId will be just fine 👍